lib/experimental/class-wp-theme-json-resolver-gutenberg.php

```diff
@@ -34,11 +34,15 @@ public static function get_theme_data( $deprecated = array(), $settings = array(
 		}
 
 		// When backporting to core, remove the instanceof Gutenberg class check, as it is only required for the Gutenberg plugin.
-		if ( null === static::$theme || ! static::$theme instanceof WP_Theme_JSON_Gutenberg ) {
+		if ( null === static::$theme || ! static::has_same_registered_blocks( 'theme' ) ) {
```

Comment:
I don't know why there is the check for `static::$theme instanceof WP_Theme_JSON_Gutenberg`, though we should aim to understand that before removing it. What was the intention behind it? Is it safe to remove it?

Reply:
It is this way in core. This just a back port.

Reply:
@spacedmonkey I've looked up git history and that check has been added in this PR https://github.com/WordPress/gutenberg/pull/42756 

As my understanding goes, without that check, there's an issue with some data not being recalculated in the plugin, so this PR has introduced a regression.

I am working on something that will fix the root issue (re: reorganizing theem.json code in the plugin), though, in the meantime, this is just such a small change that is worth adding it back until that lands.